flang/lib/Frontend/CompilerInstance.cpp

```diff
@@ -13,7 +13,6 @@
 #include "flang/Frontend/CompilerInstance.h"
 #include "flang/Frontend/CompilerInvocation.h"
 #include "flang/Frontend/TextDiagnosticPrinter.h"
-#include "flang/Parser/parsing.h"
```

Comment:
I don't think you need/want to remove these headers for this to work, and there's a couple of disadvantages to doing so: for example, clangd/other editor support might break if the right "real" includes aren't there.

These includes will get immediately skipped over in favour of the `.pch` includes anyway, because of the guard macros (`#ifndef FLANG_FRONTEND_COMPILERINSTANCE_H` etc) and the `.pch` being included first. So I think it's best to leave them in.

Reply:
FWIW clangd and other LSPs can handle this (at least mine seems fine) because they just pull from compile_commands.json. But you're right on the wider point, the the build system actually handles leaving this just fine. I'll fix the PR in a moment :)

Reply:
What happens with [`CMAKE_DISABLE_PRECOMPILE_HEADERS=ON`](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_PRECOMPILE_HEADERS.html#variable:CMAKE_DISABLE_PRECOMPILE_HEADERS)?

Also, what happens when compiling with gcc (or any other compiler), then using clangd using `compile_commands.json`?